lib/xml_builder/element.ex

```diff
@@ -0,0 +1,51 @@
+defmodule XmlBuilder.Element do
+  @typedoc """
+  The XML element internal representation.
+  """
+  @type t :: %__MODULE__{
+    name: bitstring | String.t | nil,
+    attrs: Map.t,
+    content: Tuple.t | String.t | List.t | nil}
+
+  defstruct name: nil, attrs: %{}, content: nil
+
+  def to_tuple(element), do: {element.name, element.attrs, element.content}
```

Comment:
The naming is a bit confusing `to_tuple` & `as_tuple`. Maybe it should be `to_tuple` and `to_element`?